### ISSUE TITLE
[SecurityHttp] if  don't have session,logout will throw SessionNotFoundException

### ISSUE THREAD

**Issue by mousezheng** [symfony/symfony#42258]
**Symfony version(s) affected**: 5.2.4

**Description**
If not support of sessions in the app，`getSession()`wile  throw SessionNotFoundException in the `onLogout()`

```php
class SessionLogoutListener implements EventSubscriberInterface
{
    public function onLogout(LogoutEvent $event): void
    {
        $event->getRequest()->getSession()->invalidate();
    }

    public static function getSubscribedEvents(): array
    {
        return [
            LogoutEvent::class => 'onLogout',
        ];
    }
}
```

**Possible Solution**

Check if it has session before `getSession()`

**Comment by wouterj** [comment]
:+1: Would you be open to submit a fix for the 5.2 branch?

**Comment by mousezheng** [comment]
Of course, very happy
